stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -553,6 +553,8 @@ julia> exp(A)
 """
 exp(A::StridedMatrix{<:BlasFloat}) = exp!(copy(A))
 exp(A::StridedMatrix{<:Union{Integer,Complex{<:Integer}}}) = exp!(float.(A))
+exp(A::Adjoint) = adjoint(exp(parent(A)))
+exp(A::Transpose) = transpose(exp(parent(A)))
```

Comment:
In line 554,
```
exp(A::StridedMatrix{<:BlasFloat}) = exp!(copy(A))
```
This already uses the exp! function, so shifting to this format wouldn't break consistency, I think. So, should I keep the files as is or should I modify each one to use the exp! function instead.

Reply:
Let's hear what others say. I'm inclined to keep them as they are.

Reply:
Perhaps the signature here should be `exp(A::AdjOrTransAbsMat)`, else it will catch `exp(rand(3)')`. Possibly even further restricted, for instance should this apply to `exp(view(rand(3,3), [1,3], [1,3])')` or `exp(sprand(2,2, 0.5)')`? I'm not sure.

Would vote slightly for the version which returns an ordinary matrix.

Reply:
Since `exp` is the exported function, It makes sense to me to call `exp(parent(A))`, since `exp`, not `exp!`, is more likely to be overloaded for a custom matrix type. We could also consider adding overloads for `exp!` as well, e.g.
```julia
exp!(A::Adjoint) = adjoint(exp!(parent(A)))
```
if users want to overload `exp!` for their custom matrix type. But currently `exp!` is only defined for `StridedMatrix` so this is probably not necessary.